packages/keyring-controller/src/KeyringController.ts

```diff
@@ -253,19 +253,19 @@ export class KeyringController extends BaseController<
    * @returns Promise resolving to the restored keychain object.
    */
   async createNewVaultAndRestore(password: string, seed: string | number[]) {
-    const releaseLock = await this.mutex.acquire();
-    if (!password || !password.length) {
+    const releaseLock = await this.#mutex.acquire();
+    if (!password?.length) {
       throw new Error('Invalid password');
     }
 
     try {
-      this.updateIdentities([]);
+      this.#updateIdentities([]);
       const vault = await this.#keyring.createNewVaultAndRestore(
         password,
         seed,
       );
-      this.updateIdentities(await this.#keyring.getAccounts());
-      this.fullUpdate();
+      this.#updateIdentities(await this.#keyring.getAccounts());
+      await this.fullUpdate();
```

Comment:
This potentially changes the behaviour of this function. We may want to do this instead:

```suggestion
      this.fullUpdate().catch(console.error);
```

Reply:
This makes sense to me.